### PR TITLE
Fix EXDEV error when moving tmp file.

### PR DESCRIPTION
**Pull request by bobisme**
Encountered when running verdaccio in kubernetes while using an NFS mount for storage. The error is because it's not possible to use `fs.rename` to "move" a file on this external device.

This change:
- refactors the rename logic into a function for *nix and one for windows.
- catches the EXDEV error and copies the file instead of doing a rename

Tested by deploying the change in the original environment and publishing an npm package to verdaccio.

### CODE REVIEW

**Comment by juanpicado** [comment]
Thanks @bobisme . Let me check it out.

**Comment by juanpicado** [comment]
Just some heads up, this built-in plugin is only being used with `verdaccio@3.x`.

**Comment by bobisme** [comment]
So I should also make a PR for a similar change in the 2.x branch of verdaccio?
https://github.com/verdaccio/verdaccio/blob/2.x/src/lib/storage/local/local-fs.js#L36

**Comment by juanpicado** [comment]
@bobisme let me check this one first. If we do something wrong we will impact a lot of people. I'll let you know. Thanks for the PR.

File: src/local-fs.js [hunk 1]
```diff
@@ -38,18 +38,32 @@ const tempFile = function(str) {
...
     }
-    _cb(err);
-  };
+    cb(null);
+  });
 
-  if (process.platform !== 'win32') {
-    return fs.rename(src, dst, cb);
+  function copyAndUnlink() {
+    var readStream = fs.createReadStream(src);
```
Comment (juanpicado) [inline src/local-fs.js]: `var` is not allowed here use let or const instead

File: src/local-fs.js [hunk 2]
```diff
@@ -38,18 +38,32 @@ const tempFile = function(str) {
...
+    var writeStream = fs.createWriteStream(dst);
```
Comment (juanpicado) [inline src/local-fs.js]: `var`  is not allowed here use let or const instead

**Comment by juanpicado** [comment]
btw. Ignore `eslint`  warnings

**Pull request closed without merging**